comid/integrityregisters_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package comid
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/veraison/swid"
+)
+
+func prepareRegister(t *testing.T, iType string) (*IntegrityRegisters, error) {
+	var err error
+	val := MustHexDecode(t, "e45b72f5c0c0b572db4d8d3ab7e97f368ff74e62347a824decb67a84e5224d75")
+	entry := swid.HashEntry{HashAlgID: swid.Sha256, HashValue: val}
+
+	reg := NewIntegrityRegisters()
+	for index := 0; index < 5; index++ {
+		switch iType {
+		case "uint":
+			err = reg.AddDigest(uint(index), entry)
+			require.NoError(t, err)
+		case "text":
+			i := fmt.Sprint(index)
+			err = reg.AddDigest(i, entry)
+			require.NoError(t, err)
+		default:
+			err = fmt.Errorf("invalid iType = %s", iType)
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return reg, nil
+}
+
+func TestIntegrityRegisters_AddDigest_OK(t *testing.T) {
+	_, err := prepareRegister(t, "uint")
+	require.NoError(t, err)
+}
+
+func TestIntegrityRegisters_AddDigest_NOK(t *testing.T) {
+	expectedErr := `no register to add digest`
+	register := IntegrityRegisters{}
+	err := register.AddDigest(uint(0), swid.HashEntry{})
+	assert.EqualError(t, err, expectedErr)
+
+	expectedErr = `unexpected type for index: bool`
+	var k bool
+	reg, err := prepareRegister(t, "uint")
+	require.NoError(t, err)
+	err = reg.AddDigest(k, swid.HashEntry{})
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestIntegrityRegisters_AddDigests_NOK(t *testing.T) {
+	expectedErr := `no digests to add`
+	register := IntegrityRegisters{}
+	err := register.AddDigests(uint(0), []swid.HashEntry{})
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestIntegrityRegisters_MarshalCBOR_UIntIndex_OK(t *testing.T) {
+	reg, err := prepareRegister(t, "uint")
+	require.NoError(t, err)
+	bytes, err := reg.MarshalCBOR()
+	require.NoError(t, err)
+	fmt.Printf("CBOR Payload = %x", bytes)
```

Comment:
rather than printing to stdout, the test should check that the expected value is produced

Reply:
@thomas-fossati : The order in the `map` is not always the same, hence one cannot ascertain `fixed` sequence to compare against to. If we want fixed ordering, I will have to change the implementation to encode it in a fixed manner!

Let me know, what are your views on this, shall we change internal implementation, though in the CoRIM specification we stated that the Register Order is not guaranteed when comparing..

Reply:
> @thomas-fossati : The order in the `map` is not always the same, hence one cannot ascertain `fixed` sequence to compare against to. If we want fixed ordering, I will have to change the implementation to encode it in a fixed manner!

`cbor.SortCoreDeterministic` should do the trick

> Let me know, what are your views on this, shall we change internal implementation, though in the CoRIM specification we stated that the Register Order is not guaranteed when comparing..

Sure, and the reason for that is the spec should not constrain the implementations.  So that implementations can choose whether to do deterministic encoding or not depending on their tradeoffs.

Reply:
Sure, I will check. How about JSON ? I will check, but I hope there should be a similar determinism for JSON Encoding?

Reply:
> How about JSON ? I will check, but I hope there should be a similar determinism for JSON Encoding?

That is not needed because [assert.JSONEq](https://github.com/stretchr/testify/blob/v1.8.4/assert/assertions.go#L1710) does not care about ordering of map entries.